terraform/graylog/resource_stream.go

```diff
@@ -139,21 +184,74 @@ func resourceStreamRead(d *schema.ResourceData, m interface{}) error {
 
 func resourceStreamUpdate(d *schema.ResourceData, m interface{}) error {
 	config := m.(*Config)
-	cl, err := client.NewClient(
-		config.Endpoint, config.AuthName, config.AuthPassword)
+	cl, err := client.NewClient(config.Endpoint, config.AuthName, config.AuthPassword)
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to create http client: %v", err)
 	}
-	stream, err := newStream(d)
-	if err != nil {
+
+	stream := newStream(d)
+	if err := updateStreamRule(cl, stream); err != nil {
 		return err
 	}
-	if _, err := cl.UpdateStream(stream); err != nil {
+
+	if _, err = cl.UpdateStream(stream); err != nil {
 		return err
 	}
 	return nil
 }
 
+func updateStreamRule(cl *client.Client, stream *graylog.Stream) error {
+	streamRules, _, _, err := cl.GetStreamRules(stream.ID)
+	if err != nil {
+		return fmt.Errorf("failed to get stream rules: %v", err)
+	}
+
+	// check if the stream rule was removed from the tf config
+	for _, sr := range streamRules {
+		exists := false
+		for _, r := range stream.Rules {
+			if r.ID == sr.ID {
```

Comment:
`sr.ID` is always empty because `expandStreamRule` doesn't set id.

https://github.com/suzuki-shunsuke/go-graylog/pull/14/files#diff-9f2d782424220e878e36e231eb19d6a3R125